server/game/abilities/keyword/RestoreAbility.ts

```diff
@@ -13,6 +13,7 @@ export class RestoreAbility extends TriggeredAbility {
         return {
             title: `Restore ${restoreAmount}`,
             when: { onAttackDeclared: (event, context) => event.attack.attacker === context.source },
+            locationFilter: WildcardLocation.AnyArena,
```

Comment:
What does adding the AnyArena accomplish here?

Reply:
It's because of new requirements around leader abilities. To reduce the possibility of mixing up leader and leader unit side abilities, there's now a requirement that any ability which is attached to a leader explicitly declare its active locations instead of using the default.

This gets handled automatically for abilities that are written in the .ts file. The side effect is that keyword abilities bypass that. So we either need to refactor to try and make an exception for keyword abilities or just make sure to explicitly apply a `locationFilter` property for triggered keyword abilities.